### DISCUSSION TITLE
PS3 Eye Camera for IR (Infrared) use

### DISCUSSION THREAD

**Discussion by Borbonjuggler** [jyjblrd/Mocap-Drones#10] [General]
Hey there,

I'm planning to adapt your code for tracking movements in a tangible interface unit. However, However, PS3 Eye Cameras seem to be hard to come by in Austria. The combined cost of the lens and camera can be around 20 euros, so I was considering alternatives like [this Raspberry Pi infrared USB camera from Amazon](https://www.amazon.de/-/en/DFROBOT-Raspberry-Recognition-Distance-Learning/dp/B08FJ6CHDT/ref=sr_1_2?crid=2OL2IHFT73MPC&keywords=raspberry+pi+infrared+usb&qid=1700693503&sprefix=raspberry+pi+infrared+usb%2Caps%2C147&sr=8-2). However, you've mentioned synchronizing shutters, which might be a critical factor. Could you suggest any other cameras that would be suitable for this project? Additionally, if I manage to get a couple of PS3 cameras, would [this lens from Amazon](https://www.amazon.de/-/en/Tangxi-Surveillance-Camera-Angle-Fisheye-Default/dp/B083NWC113/ref=sr_1_102?crid=1IYSUV0O28LT1&keywords=%22night+vision%22+security+camera+lenses&qid=1700691433&sprefix=night+vision%22+security+camera+lenses%2Caps%2C126&sr=8-102) be suitable?

**Comment by CleisonManriqueAguirre** [comment]
So then you need two raspberries ? each one for each camera . Is it possible to use espcam ?

**Reply by jyjblrd** [reply]
I did consider using espcams, but it will probably have far too much latency if you are streaming the data from the esp32 to a computer.

**Reply by Borbonjuggler** [reply]
That would be nice, ESP32-CAM Video Streaming. I saw in the code you config the camera to 90fps. Sadly with the ESPcamera people has reported that they can get between 12 - [50fps ](https://www.reddit.com/r/esp32/comments/v2a0k6/comment/iaslhot/?utm_source=share&utm_medium=web2x&context=3)HD, after propping [shielding](https://www.reddit.com/r/esp32/comments/r9g5jc/fixing_ymmv_the_poor_frame_rate_on_the_esp32cam/)

**Comment by jyjblrd** [comment]
If you want a camera with synchronization, the OV9281 camera sensor has an external trigger pin. [This USB webcam](https://www.arducam.com/product/arducam-120fps-global-shutter-usb-camera-board-1mp-720p-ov9281-uvc-webcam-module-with-low-distortion-m12-lens-without-microphones-for-computer-laptop-android-device-and-raspberry-pi/) version of it has a sync pin, with more details in the [documentation](https://docs.arducam.com/UVC-Camera/Appilcation-Note/External-Trigger-Mode/OV9281-Global-Shutter/#13-external-trigger-function-instructions). This is the most promising alternative I've found, and it might be better than the ps3 eye camera (but more expensive).

If you don't need synchronization, any usb webcam should work. I wouldn't worry about the synchronization unless you are going to be moving with very fast moving objects at high frame rates, but yeah I don't think the camera you linked would be able to be synchronized. 

That lens would work, you would just need to design a mount to attach it to the ps3 camera and 3D print it, which should be easy to do.

**Reply by Borbonjuggler** [reply]
I found some PS3 Eye cameras online today for about 13 euros each, including shipping. Now, I'm waiting for the sellers to confirm if they're still available and interested in selling. Once I get their response, I'll proceed to buy the lenses. thanks for you advice

**Comment by TianmingLi** [comment]
Hey there,

Based on the above discussion,

I found this camera on [Taobao](https://item.taobao.com/item.htm?id=649833197106#detail),

Its sensor is OV9281, with built-in infrared-pass lens and infrared lights.

Is it a good alternative to Ps3 Eye?


https://github.com/jyjblrd/Mocap-Drones/assets/5005440/448171f2-2786-48f1-b9ea-fa266d51b74d

![image](https://github.com/jyjblrd/Mocap-Drones/assets/5005440/3258e6e0-8ae8-4476-82f7-d5392adbc959)

**Reply by Carloslius** [reply]
Excuse me, can this camera be used for this project?

**Reply by jyjblrd** [reply]
Yeah probably, any camera will work as long as it can be plugged in as a webcam. You may need to edit the code slightly to connect it to this webcam instead of the ps3 eye cameras. You will also need to calculate the camera's intrinsics.

**Reply by boba1234567890** [reply]
just wondering where would that be in the code 
i cant find where i need to change the code to get a different webcam to work 

thanks
